src/components/DigitalConverterComponent.cpp

```diff
@@ -0,0 +1,62 @@
+#include <stdlib.h>
+#include <time.h>
+
+#include <hyro/DigitalConverterComponent.h>
+
+using namespace hyro;
+
+namespace signal_generator
+{
+std::shared_ptr<hyro::HyroLogger> DigitalConverterComponent::s_logger = hyro::HyroLoggerManager::CreateLogger("DigitalConverterComponent");
+
+hyro::Result
+DigitalConverterComponent::init (const hyro::ComponentConfiguration & config)
+{
+  signal_input = this->registerInput<SignalMsgs>("analog_input"_uri, config);
+  signal_output = this->registerOutput<float>("digital_output"_uri, config);
+
+  m_thresholding.setThreshold(m_threshold);
+  
+  return hyro::Result::RESULT_OK;
+}
+
+hyro::Result
+DigitalConverterComponent::reset ()
+{
+  // Reset all variables and objects like the object was just created
+  return hyro::Result::RESULT_OK;
+}
+
+hyro::Result
+DigitalConverterComponent::check ()
+{
+  // Check everything is OK here
```

Comment:
You need to check if the connection was established with success. See the example below:

```c++
  if (m_input->connectionStatus() != ConnectionStatus::CONNECTION_UP)
    return Result::RESULT_ERROR;
```